wicket-core/src/main/java/org/apache/wicket/ajax/AbstractDefaultAjaxBehavior.java

```diff
@@ -106,8 +107,7 @@ public void renderHead(final Component component, final IHeaderResponse response
 		RequestCycle requestCycle = component.getRequestCycle();
 		Url baseUrl = requestCycle.getUrlRenderer().getBaseUrl();
 		CharSequence ajaxBaseUrl = Strings.escapeMarkup(baseUrl.toString());
-		response.render(JavaScriptHeaderItem.forScript("Wicket.Ajax.baseUrl=\"" + ajaxBaseUrl
-			+ "\";", "wicket-ajax-base-url"));
```

Comment:
Mh I didn't look into the implementation but is wicket-ajax-base-url not used to be applied to the OnLoadHeaderItem, too. I could imagine that the information gets lost in this case.

Reply:
Ah this is only for the id attribute - so ignore this comment.

Reply:
this missing `id` attribute breaks `org.apache.wicket.util.tester.BaseWicketTester.getWicketAjaxBaseUrlEncodedInLastResponse()` ....
Will set it back if there are no better solutions ...

Reply:
It's not missing, it's just on a different location (see below), where getWicketAjaxBaseUrlEncodedInLastResponse() doesn't find it - we have to change that method accordingly.

Reply:
previous code set the id attribute, so the check was trivial (node.attr === 'wicket-ajax-base-url'). Current implementation adds text content. which have to be parsed, maybe it worth to set attribute back to avoid parsing the text?

Reply:
Setting the baseUrl has to be deferred until onLoad, otherwise the deferred wicket-ajax-jquery.js hasn't been loaded yet and Wicket.Ajax is not defined.